src/packet.rs

```diff
@@ -180,12 +180,22 @@ pub enum BlobError {
 pub struct Recycler<T> {
     #[cfg_attr(feature = "cargo-clippy", allow(type_complexity))]
     gc: Arc<Mutex<Vec<(Arc<RwLock<T>>, &'static str)>>>,
+    allocated_count: Arc<AtomicUsize>,
+    recycled_count: Arc<AtomicUsize>,
+    reuse_count: Arc<AtomicUsize>,
+    skipped_count: Arc<AtomicUsize>,
+    name: String,
 }
 
 impl<T: Default> Default for Recycler<T> {
```

Comment:
This is what I'd imagine the default implementation would look like. Did you try adding `#[derive(Clone)]` to the struct?

Reply:
Just tried..

```
error[E0599]: no method named `clone` found for type `packet::Recycler<packet::Packets>` in the current scope
   --> src/rpu.rs:57:29
    |
57  |             packet_recycler.clone(),
    |                             ^^^^^
    |
   ::: src/packet.rs:225:1
    |
225 | pub struct Recycler<T> {
    | ---------------------- method `clone` not found for this
    |
    = note: the method `clone` exists but the following trait bounds were not satisfied:
            `packet::Recycler<packet::Packets> : std::clone::Clone`
    = help: items from traits can only be used if the trait is implemented and in scope
    = note: the following trait defines an item `clone`, perhaps you need to implement it:
            candidate #1: `std::clone::Clone`
```

Reply:
oh. maybe i need `use std::clone::Clone` in those files...

Reply:
hmmm. no that wasn't it.